### PR TITLE
Remove offset-based C-ext interop fallbacks

### PR DESCRIPTION
**Pull request by eregon**
* Sulong always uses struct names for interop now.

Let's clean some code :) cc @rschatz

### CODE REVIEW

File: lib/truffle/truffle/cext.rb [hunk 1]
```diff
@@ -20,22 +20,10 @@ def initialize(data)
   end
 
   class RData
-    DATA_FIELD_INDEX = 2
```
Comment (nirvdrum) [inline lib/truffle/truffle/cext.rb]: Hooray!

**Comment by eregon** [comment]
Unfortunately a bug needs to be addressed in Sulong first for this to be merged.

**Comment by eregon** [comment]
This is blocked by GR-8014

**Comment by chrisseaton** [comment]
This is still a goal, but it will need to be heavily rebased or re-done due to changes to the interop API.

**Comment by eregon** [comment]
OK, I'll close this then and leave it to you @chrisseaton.
I think it's just easier to reapply the changes manually since the diff is small and obvious, so rebasing this is not worthwhile.

**Pull request closed without merging**